fw/http_parser.c

```diff
@@ -1249,12 +1249,18 @@ __FSM_STATE(st_curr) {							\
 __FSM_STATE(RGen_HdrOtherN) {						\
 	__FSM_MATCH_MOVE(token, RGen_HdrOtherN, 0);			\
 	if (likely(*(p + __fsm_sz) == ':')) {				\
+		size_t n = 0;						\
+		if (TFW_STR_EMPTY(&parser->hdr)) {			\
+			n = p - (unsigned char *)parser->hdr.data;	\
+		} else {						\
+			n = parser->hdr.len + __data_off(p);		\
+		}							\
```

Comment:
We need `n` only to not to get into the `if` statement at the below, so the `else` branch is senseless and this `if` condition can be aggregated with the below one. Moreover, we don't need `n` value, just need to for 0 or not, i.e. this means "whether we saw at least one character in the header?". The last thing is that we really can use `parser` and don't need `&msg->stream->parser.hdr`. I propose to rewrite the fix as
```C
--- a/fw/http_parser.c
+++ b/fw/http_parser.c
@@ -1252,10 +1252,11 @@ __FSM_STATE(RGen_HdrOtherN) {                                           \
                /*                                                      \
                 * Header name must contain at least one character, but \
                 * ':' can be found at the beginning of a new skb or    \
-                * fragment, it's ok.                                   \
+                * fragment, it's ok. We also might get into the state  \
+                * either on the fist character of the header or not.   \
                */                                                      \
-               if (unlikely(!__fsm_sz                                  \
-                            && TFW_STR_EMPTY(&msg->stream->parser.hdr))) \
+               if (unlikely(!__fsm_sz && TFW_STR_EMPTY(parser->hdr)    \
+                            && p != (unsigned char *)parser->hdr.data))\
                {                                                       \
                        TFW_PARSER_BLOCK(RGen_HdrOtherN);               \
                }                                                 
```